service/controller/chart/v1/resource/release/resource.go

```diff
@@ -19,6 +18,17 @@ const (
 	helmDeployedStatus = "DEPLOYED"
 )
 
+var (
+	// releaseTransitionStatuses is used to determine if the Helm Release is
+	// currently being updated.
+	releaseTransitionStatuses = []string{
+		"DELETING",
```

Comment:
I remember to have that seen already recently somewhere else. IIRC @TheoBrigitte worked on something pretty similar. Could we align this or have a common lib for such things?

Reply:
Yes this is based on @TheoBrigitte's change but for the new CRD. The reason we want this is because we want to reconcile failed helm releases by using `helm upgrade --force`.

The problem is `--force` is not safe in all situations. We can't use it for Ingress Controller for example because it deletes and recreates the k8s resources. We will control its usage via an annotation.

So Theo started this and I will finish it in later PRs. As it makes sense for Batman to get this sorted.

Reply:
I would be nice to have this code which is common with chartconfig resource defined in a single place. I don't mind if its a package or a lib.

Reply:
The chart CRD support is basically done. It needs this PR and the status resource which I'm working on.

The chartconfig CRD is then deprecated and I'll update the readme. So I don't want to extract logic that will be deleted. If we need important fixes for the current chartconfig CRs we can backport them.